extensions/azurecore/src/azureResource/services/terminalService.ts

```diff
@@ -168,13 +168,19 @@ class AzureTerminal implements vscode.Pseudoterminal {
 		this.socket.removeAllListeners('close');
 
 		this.socket.terminate();
+		this.socket.close();
```

Comment:
Why are you adding this? Terminate will effectively close the socket (it just does it immediately) so this seems at best unnecessary to have both.

Reply:
sockets should always be closed, I'm not sure what terminate does. This was a bug if anything because when I was testing for hours I eventually errored out after 20 open powershell saying I had reached the limit

Reply:
So first off if you're making fixes like this please always either put them in a separate PR or call it out in the description. Otherwise it's confusing/easy to miss.

Anyways so that's interesting, looking at the source the main difference seems to be that terminate destroys the socket immediately while close sends a closing frame first to tell the other end that it's closing

https://github.com/websockets/ws/blob/master/lib/websocket.js#L369
https://github.com/websockets/ws/blob/master/lib/websocket.js#L220

So I'm guessing what's happening here is that because we were just terminating it the server was still keeping its side open and thus you hit the limit. Calling close likely tells it to clean up the connection so avoids that issue. So I'd say that we should just call close and not call terminate - calling both as mentioned is unnecessary and could cause conflicts (for example there's probably a race condition here now where the socket could be destroyed before we try to send the close frame so it'll either do nothing or possibly error out)